api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptDao.java

```diff
@@ -0,0 +1,30 @@
+package org.pmiops.workbench.cdr.dao;
+import org.pmiops.workbench.cdr.model.Concept;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface ConceptDao extends CrudRepository<Concept, Long> {
+
+    @Query(value = "select c.* from cdr.concept c where c.concept_name like %:conceptName% and c.domain_id in ('Condition','Observation','Procedure', 'Measurement', 'Drug') order by c.count_value desc limit 10;",
+            nativeQuery = true)
+    List<Concept> findConceptLikeName(@Param("conceptName") String conceptName);
+
+    @Query(value = "select c.* from cdr.concept c where c.domain_id in ('Condition','Observation','Procedure', 'Measurement', 'Drug') order by c.count_value desc limit 10;",
+    nativeQuery = true)
+    List<Concept> findConceptsOrderedByCount();
+
+    @Query(value = "select c.* from cdr.concept c join cdr.concept_relationship rel on rel.concept_id_1 = c.concept_id and rel.concept_id_2 = :conceptId and rel.relationship_id = 'maps to' where c.concept_id != :conceptId order by c.count_value desc",
+            nativeQuery = true)
+    List<Concept> findConceptsMapsToChildren(@Param("conceptId") long conceptId);
+
+    @Query(value = "select c.* from cdr.concept c join cdr.concept_relationship rel on rel.concept_id_2 = c.concept_id and rel.concept_id_1 = :conceptId and rel.relationship_id = 'maps to' where c.concept_id != :conceptId order by c.count_value desc",
+            nativeQuery = true)
+    List<Concept> findConceptsMapsToParents(@Param("conceptId") long conceptId);
+
+    List<Concept> findByConceptName(String conceptName);
```

Comment:
Can we write unit test for this Dao?

Reply:
Talked with Dan. Will add tests very soon after this request gets in.